server/src/main/java/com/linecorp/centraldogma/server/internal/api/RepositoryServiceV1.java

```diff
@@ -307,6 +315,117 @@ public CompletableFuture<RepositoryDto> updateStatus(Project project,
                   .thenApply(unused -> DtoConverter.convert(repository, newStatus));
     }
 
+    /**
+     * POST /projects/{projectName}/repos/{repoName}/migrate/encrypted
+     *
+     * <p>Migrates the repository to an encrypted repository.
+     */
+    @Post("/projects/{projectName}/repos/{repoName}/migrate/encrypted")
+    @RequiresSystemAdministrator
+    public CompletableFuture<RepositoryDto> migrateToEncryptedRepository(ServiceRequestContext ctx,
+                                                                         Project project,
+                                                                         Repository repository,
+                                                                         Author author) {
+        validateMigrationPrerequisites(ctx, project, repository);
+        ctx.setRequestTimeoutMillis(Long.MAX_VALUE); // Disable the request timeout for migration.
```

Comment:
Just out of curiosity, how long did it take to migrate a repository having 1k commits?

Reply:
Thanks for the input. It may take a couple of minutes when processing real data.